bsp/tm4c123bsp/libraries/Drivers/drv_uart.c

```diff
@@ -0,0 +1,335 @@
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "hw_memmap.h"
+#include "hw_ints.h"
+#include "sysctl.h"
+#include "gpio.h"
+#include "pin_map.h"
+#include "interrupt.h"
+#include "rom_map.h"
+#include "uart.h"
+
+#include "drv_uart.h"
+#include "uart_config.h"
+#include "tm4c123_config.h"
+
+#ifdef RT_USING_SERIAL
+
+#define   LOG_TAG  "drv.uart"
+#include <drv_log.h>
+
+#if !defined(BSP_USING_UART0)&&!defined(BSP_USING_UART1)&&!defined(BSP_USING_UART2)&&!defined(BSP_USING_UART3)
+    #error "Please define at least one BSP_USING_UARTx"
+#endif
+
+enum {
+	
+#ifdef BSP_USING_UART0
+	UART0_INDEX,
+#endif
+	
+#ifdef BSP_USING_UART1
+	UART1_INDEX,
+#endif
+
+#ifdef BSP_USING_UART2
+	UART2_INDEX,
+#endif
+	
+#ifdef BSP_USING_UART3
+	UART3_INDEX,
+#endif	
+};
+
+uint32_t  uart_intbase[]=
+{
+#ifdef BSP_USING_UART0
+	  INT_UART0,
+#endif
+	
+#ifdef BSP_USING_UART1
+	  INT_UART1,
+#endif
+
+#ifdef BSP_USING_UART2
+	  INT_UART2,
+#endif
+	
+#ifdef BSP_USING_UART3
+	  INT_UART3
+#endif		 
+};
+
+static struct tm4c123_uart_config uart_config[] =
+{
+#ifdef BSP_USING_UART0
+    UART0_CONFIG,
+#endif
+	
+#ifdef BSP_USING_UART1
+    UART1_CONFIG,
+#endif
+
+#ifdef BSP_USING_UART2
+    UART2_CONFIG,
+#endif
+	
+#ifdef BSP_USING_UART3
+    UART3_CONFIG,
+#endif	
+};
+
+static struct tm4c123_uart uart_obj[sizeof(uart_config) / sizeof(uart_config[0])] = {0};
+
+
+//struct serial_configure
```

Comment:
If these codes are unnecessary, remove them.

Reply:
OK. I've modified  the entire code to check and correct these issues .